Bottom: e8b02f669338a20b62a01e695ac4ec1db4eb8eae
Top:    bf1e39369f3939ab6acb242737f2b25da2c8b83e
Author: Jason Wang <jasowang@redhat.com>
Date:   2018-01-24 16:18:03 +0800

debug


---

diff --git a/drivers/vhost/vhost.c b/drivers/vhost/vhost.c
index 0b19845..5a06bb3 100644
--- a/drivers/vhost/vhost.c
+++ b/drivers/vhost/vhost.c
@@ -1997,11 +1997,17 @@ static int vhost_read_indices(struct vhost_virtqueue *vq, u16 num)
 	int ret, ret2;
 	int i;
 
-	BUG_ON(indices->read_tail != indices->tail);
+	//BUG_ON(indices->read_tail != indices->tail);
+	printk("read indices tail %d read_tail %d head %d\n",
+		indices->tail, indices->read_tail, indices->head);
+	if (indices->read_tail != indices->tail)
+		printk("read_tail is not equal to tail\n");
 
 	if (unlikely(vhost_get_avail(vq, avail_idx, &vq->avail->idx))) {
 		vq_err(vq, "Failed to access avail idx at %p\n",
 		       &vq->avail->idx);
+		printk("Failed to access avail idx at %p\n",
+			&vq->avail->idx);
 		return -EFAULT;
 	}
 	last_avail_idx = vq->last_avail_idx & (vq->num - 1);
@@ -2013,12 +2019,15 @@ static int vhost_read_indices(struct vhost_virtqueue *vq, u16 num)
 		ret2 = vhost_get_avail(vq, heads[i],
 				      &vq->avail->ring[last_avail_idx]);
 		if (unlikely(ret2)) {
+			printk("Failed to get descriptor\n")
 			vq_err(vq, "Failed to get descriptors\n");
 			return -EFAULT;
 		}
 		if (unlikely(heads[i] >= vq->num)) {
 			vq_err(vq, "Guest says index %u > %u is available",
 			       heads[i], vq->num);
+			printk("Guest says index %u > %u is available",
+			       heads[i], vq->num);
 			return -EINVAL;
 		}
 		last_avail_idx = (last_avail_idx + 1) & (vq->num - 1);
@@ -2050,6 +2059,9 @@ static int vhost_read_descs(struct vhost_virtqueue *vq, int num)
 			vq_err(vq, "Failed to get descriptor: "
 				"idx %d addr %p\n",
 				head, vq->desc + head);
+			printk("Failed to get descriptor: "
+				"idx %d addr %p\n",
+				head, vq->desc + head);
 			goto err;
 		}
 		descs->head++;
@@ -2076,6 +2088,9 @@ static int vhost_read_descs(struct vhost_virtqueue *vq, int num)
 				vq_err(vq, "Failed to get descriptor: "
 					   "idx %d addr %p\n",
 					   head, vq->desc + head);
+				printk("Failed to get descriptor: "
+					"idx %d addr %p\n",
+					head, vq->desc + head);
 				goto err;
 			}
